lib/utils/visUtil.hpp

```diff
@@ -527,6 +528,110 @@ class movingAverage {
     bool initialised = false;
 };
 
+/**
+ * @class SlidingWindowMinMax
```

Comment:
I think class is is based off this [work](https://www.nayuki.io/page/sliding-window-minimum-maximum-algorithm) correct?  If so, you should include a reference to the original blog post/code where the original ideas/code come from.